### ISSUE TITLE
Does Postgres Adapter display the correct error when db exists?

### ISSUE THREAD

**Issue by wjdhamilton** [hanami/model#322]
Hello everyone!

When running `hanami db create` and there is already a database in existence, the error returned is as follows: 

```
createdb: database creation failed. There is 1 other session using the database.
```

A quick look at the code in model/lib/hanami/model/migrator/postgres_adapter.rb reveals the following snippet: 

``` ruby

message = if error_message.match(/already exists/)
              "createdb: database creation failed. There is 1 other session using the database."
```

Which implies to me that the error message is the same regardless of whether or not there really _is_ another 1 (or many more) sessions using the database. 

I would have expected that the app would differentiate between contexts when another connection is present and when it is simply the case that the db has already been created. Shouldn't the error message simply say that the db already exists if there are no sessions using the db?

**Comment by cllns** [comment]
Thanks for reporting this!
This is a really good point. We definitely want accurate and helpful error messages.

Right now, @jodosha is working on re-working the back-end of `hanami-model` to be based on [Ruby Object Mapper (`rom`)](https://github.com/rom-rb/rom) and I *think* this code will be going away. 
So, I don't think it's worth creating a PR, but @jodosha can correct me if I'm wrong :)

**Comment by wjdhamilton** [comment]
No problem! Glad to have been of service. I'll await @jodosha 😄

**Comment by jodosha** [comment]
@wjdhamilton I think that this is fine to be fixed in master first **with the current code**. So the `new-engine` branch will already have working code.

@wjdhamilton or @cllns Can you please open a PR that fix this problem? It should target `master` branch, of course. Thanks! 👍

**Comment by cllns** [comment]
@wjdhamilton Want to take this one? If not, I can do it. Feel free to ping me in the [Hanami Gitter chat](https://gitter.im/hanami/chat) if you need any help :)

**Comment by wjdhamilton** [comment]
Yes, I’ll take this one. I will probably have to ask you a few questions first but I’ll get on it later this week.

**Comment by tejasbubane** [comment]
I guess this can be closed.

**Comment by jodosha** [comment]
Fixed by #325
